clients/drcachesim/tracer/raw2trace.cpp

```diff
@@ -878,6 +861,103 @@ raw2trace_t::do_conversion()
     return "";
 }
 
+const instr_summary_t *
+raw2trace_t::get_instr_summary(uint64 modidx, uint64 modoffs, INOUT app_pc *pc,
+                               app_pc orig)
+{
+    const app_pc decode_pc = *pc;
+    const instr_summary_t *ret =
+        static_cast<const instr_summary_t *>(hashtable_lookup(&decode_cache, decode_pc));
+    if (ret == nullptr) {
+        instr_summary_t *desc = new instr_summary_t();
+        if (!instr_summary_t::construct(dcontext, pc, orig, desc, verbosity)) {
+            WARN("Encountered invalid/undecodable instr @ %s+" PFX,
+                 modvec()[static_cast<size_t>(modidx)].path, (ptr_uint_t)modoffs);
+            return nullptr;
+        }
+        hashtable_add(&decode_cache, decode_pc, desc);
+        ret = desc;
+    } else {
+        // TODO(mtrofin): log some rendering of the instruction summary that will be
```

Comment:
style: "XXX i#NNN: " where NNN is the issue number

Reply:
done